app/connector/api-provider/src/main/resources/META-INF/syndesis/connector/api-provider.json

```diff
@@ -64,266 +70,64 @@
-                    "label": "501 Not Implemented",
-                    "value": "501"
-                  },
-                  {
-                    "label": "502 Bad Gateway",
-                    "value": "502"
-                  },
-                  {
-                    "label": "503 Service Unavailable",
-                    "value": "503"
-                  },
-                  {
-                    "label": "504 Gateway Timeout",
-                    "value": "504"
-                  },
-                  {
-                    "label": "505 HTTP Version Not Supported",
-                    "value": "505"
-                  },
-                  {
-                    "label": "506 Variant Also Negotiates",
-                    "value": "506"
-                  },
-                  {
-                    "label": "507 Insufficient Storage",
-                    "value": "507"
-                  },
-                  {
-                    "label": "508 Loop Detected",
-                    "value": "508"
-                  },
-                  {
-                    "label": "510 Not Extended",
-                    "value": "510"
-                  },
-                  {
-                    "label": "511 Network Authentication Required",
-                    "value": "511"
-                  }
-                ],
                 "javaType": "String",
                 "kind": "parameter",
                 "required": true,
                 "secret": false,
-                "type": "string"
+                "type": "select"
+              },
+              "errorHandling": {
+                "order": 2,
+                "componentProperty": false,
+                "deprecated": false,
+                "displayName": "Error Handling",
+                "javaType": "String",
+                "kind": "parameter",
+                "required": false,
+                "secret": false,
+                "type": "legend"
+              },
+              "returnBody": {
+                "order": 3,
+                "componentProperty": false,
+                "deprecated": false,
+                "displayName": "Include error message in the return body",
+                "displayNameCheckBox": "Select return body preference",
+                "javaType": "Boolean",
+                "kind": "parameter",
+                "required": false,
+                "secret": false,
+                "type": "boolean"
+              },
+              "errorResponseCodes": {
+                "order": 4,
+                "componentProperty": false,
+                "deprecated": false,
+                "defaultValue": "{}",
+                "description": "The return code to set according to different error situations",
+                "displayName": "Error Response Codes",
+                "javaType": "Map",
+                "kind": "parameter",
+                "required": false,
+                "secret": false,
+                "type": "mapset"
```

Comment:
UI knows about `mapset`?

Reply:
Yep ask Stan if you want to know more about it.